cacheops/redis.py

```diff
@@ -29,24 +30,48 @@ def handle_connection_failure(call):
 
 client_class_name = getattr(settings, 'CACHEOPS_CLIENT_CLASS', None)
 client_class = import_string(client_class_name) if client_class_name else redis.StrictRedis
+redis_replicas = []
+
+
+def ip(url):
+    return socket.gethostbyname(urlparse(url).hostname)
+
+
+def set_redis_replicas():
+    # the conf could be a list or string
+    # list would look like: ["redis://cache-001:6379/1", "redis://cache-002:6379/2"]
+    # string would be: "redis://cache-001:6379/1,redis://cache-002:6379/2"
+    global redis_replicas
+    if isinstance(settings.REDIS_REPLICAS, six.string_types):
+        temp = settings.REDIS_REPLICAS.split(',')
+    else:
+        temp = list(settings.REDIS_REPLICAS)
+    primary_url = settings.REDIS_MASTER
+    primary_ip = ip(primary_url)
+    temp = [r for r in temp if ip(r) != primary_ip]
+    temp = map(redis.StrictRedis.from_url, temp)
+    temp = [r for r in temp for _ in range(settings.REDIS_REPLICA_WEIGHT)]
```

Comment:
These three lines are insanity distilled in code.  Can I trouble you to please add some comments to talk about what's happening here?